src/main/java/duke/task/Ui.java

```diff
@@ -0,0 +1,139 @@
+                "\\ \\/ /|_  /_  /_  /\n" +
+                " >  <  / / / / / / \n" +
+                "/_/\\_\\/___/___/___|";
+        System.out.println(logo);
+        System.out.println("Hello uwu! I'm xzzz,");
+        System.out.println("You can check your schedwle here (ɔ◔‿◔)ɔ ♥!");
+    }
+
+    /**
+     * Prints the goodbye message.
+     */
+    public void goodbye() {
+        System.out.println("Cya later~ ≧◉◡◉≦");
+    }
+
+
+    /**
+     * Prints the list for display.
+     * @param toDoList List to be displayed.
+     */
+    public void displayList(ArrayList<Task> toDoList) {
+        int number = 1;
+        System.out.println("here are your tasks ☜(ˆ▿ˆc)");
+        for (Task item : toDoList) {
+            System.out.println(number + ". " + item.toString());
+            number++;
+        }
+    }
+
+    /**
+     * Print the error caused by illegal argument.
+     * @param ex IllegalArgument error.
+     */
+    public void showIllegalArgumentError(Exception ex) {
+        System.out.println(ex);
+    }
+
+    /**
+     * Print the error caused by incomplete argument.
+     */
+    public void showIncompleteArgumentError() {
+        System.out.println("UH-OH!! you gotta fill in the description and deadline date " +
+                "/ event date to create a valid task (> <ლ)");
+    }
+
+    /**
+     * Prints the confirmation for addition of a <code>Task</code> to the <code>toDoList</code>.
+     * @param newTask Task to be added.
+     * @param toDoList List for the task to be added to.
+     */
+    public void confirmAddition(Task newTask, ArrayList<Task> toDoList) {
+        System.out.println("okie!! (✿◠‿◠)  i have added: \n" +
+                newTask + "\n" +
+                "now there are " + toDoList.size() + " tasks in the list! get to work (ง︡'-'︠)ง");
+    }
+
+    /**
+     * Prints a message to indicate the bot does not understand the command.
+     */
+    public void doNotUnderstand() {
+        System.out.println("sowwy i don't understand what that means ಠ_ಥ try something else pwease??");
+    }
+
+
+    /**
+     * Prints a confirmation that a <code>Task</code> at the index in the list has been marked as done.
+     * @param toDoList List of tasks.
+     * @param idx Index of the task that has been marked.
+     */
+    public void markAsDone(ArrayList<Task> toDoList, int idx) {
+        System.out.println("yay!!! this task is now marked as done ٩(˘◡˘)۶");
+        System.out.println(toDoList.get(idx - 1).toString());
+    }
+
+    /**
+     * Prints a confirmation that a <code>Task</code> at the index in the list has been marked as undone.
+     * @param toDoList List of tasks.
+     * @param idx Index of the task that has been unmarked.
+     */
+    public void unmarkAsDone(ArrayList<Task> toDoList, int idx) {
+        System.out.println("this task is now marked as not done yet... do it soon! ᕙ(`▿´)ᕗ");
+        System.out.println(toDoList.get(idx - 1).toString());
+    }
```

Comment:
Perhaps I missed it out while reading your code, but was the possibility that the user enters an invalid task number for mark/unmark considered? (e.g. 3 tasks in the list but user enters mark 10).

Reply:
Hmm... You bring up a good point! I forgot to consider cases where the idx exceeds the size or when the idx is negative. Thanks!